### ISSUE TITLE
Is this a bug or me?  Failure during initial ansible playbook build, cannot find file

### ISSUE THREAD

**Issue by sobriant74** [spantaleev/matrix-docker-ansible-deploy#2359]
Installing the playbook for the first time on a new ubuntu 22 server, via ansible from another ubuntu server, using the devture/ansible and I cannot get past the "Ensure matrix network is created in Docker" step. Just says file not found. I'm following the install guide and connecting between servers via SSH creds as a user and then raising that user to root via sudo. Command I'm running: sudo ansible-playbook -i inventory/hosts setup.yml --tags=install-all,ensure-matrix-users-created,start --ask-pass 
Is this a file from the playbook I copied to the Ansible server with git, or is it referencing a prereq that got missed on the host server for Matrix? It looks like its something missing from Python3, but thats installed on both the control node and the managed node. my hosts file on the ansible control server: 
[matrix_servers]
chat.domainname.com ansible_host=10.101.40.231 `ansible_ssh_user=username become=true become_user=root ansible_python_interpreter=/usr/bin/python3

Is this a bug or am I missing something on my end?
Any help is appreciated.
Thanks

TASK [custom/matrix-base : Ensure Matrix network is created in Docker] ********************************************************************
task path: /home/username/matrix-docker-ansible-deploy/roles/custom/matrix-base/tasks/setup_matrix_base.yml:14
Using module file /usr/lib/python3/dist-packages/ansible_collections/community/docker/plugins/modules/docker_network.py
Pipelining is enabled.
<10.101.40.231> ESTABLISH SSH CONNECTION FOR USER: None
<10.101.40.231> SSH: EXEC sshpass -d10 ssh -C -o ControlMaster=auto -o ControlPersist=60s -o ConnectTimeout=10 -o 'ControlPath="/root/.ansi                                                                     ble/cp/bfb95ee9cf"' 10.101.40.231 '/bin/sh -c '"'"'sudo -H -S -n  -u root /bin/sh -c '"'"'"'"'"'"'"'"'echo BECOME-SUCCESS-iwqafauaarajcgadh                                                                     ocqszrgpxjijqyx ; /usr/bin/python3'"'"'"'"'"'"'"'"' && sleep 0'"'"''
Escalation succeeded
<10.101.40.231> (1, b'\n{"failed": true, "msg": "Error connecting: Error while fetching server API version: (\'Connection aborted.\', FileN                                                                     otFoundError(2, \'No such file or directory\'))", "exception": "  File \\"/tmp/ansible_community.docker.docker_network_payload_44w0ssay/ans                                                                     ible_community.docker.docker_network_payload.zip/ansible_collections/community/docker/plugins/module_utils/common.py\\", line 232, in __ini                                                                     t__\\n    super(AnsibleDockerClientBase, self).__init__(**self._connect_params)\\n  File \\"/usr/local/lib/python3.10/dist-packages/docker/                                                                     api/client.py\\", line 197, in __init__\\n    self._version = self._retrieve_server_version()\\n  File \\"/usr/local/lib/python3.10/dist-pa                                                                     ckages/docker/api/client.py\\", line 221, in _retrieve_server_version\\n    raise DockerException(\\n", "invocation": {"module_args": {"nam                                                                     e": "matrix", "driver": "bridge", "docker_host": "unix://var/run/docker.sock", "api_version": "auto", "timeout": 60, "tls": false, "use_ssh                                                                     _client": false, "validate_certs": false, "debug": false, "connected": [], "state": "present", "driver_options": {}, "force": false, "appen                                                                     ds": false, "labels": {}, "tls_hostname": null, "ca_cert": null, "client_cert": null, "client_key": null, "ssl_version": null, "ipam_driver                                                                     ": null, "ipam_driver_options": null, "ipam_config": null, "enable_ipv6": null, "internal": null, "scope": null, "attachable": null}}}\n',                                                                      b'')
<10.101.40.231> Failed to connect to the host via ssh:
The full traceback is:
  File "/tmp/ansible_community.docker.docker_network_payload_44w0ssay/ansible_community.docker.docker_network_payload.zip/ansible_collectio                                                                     ns/community/docker/plugins/module_utils/common.py", line 232, in __init__
    super(AnsibleDockerClientBase, self).__init__(**self._connect_params)
  File "/usr/local/lib/python3.10/dist-packages/docker/api/client.py", line 197, in __init__
    self._version = self._retrieve_server_version()
  File "/usr/local/lib/python3.10/dist-packages/docker/api/client.py", line 221, in _retrieve_server_version
    raise DockerException(
fatal: [chat.servername.com]: FAILED! => changed=false
  invocation:
    module_args:
      api_version: auto
      appends: false
      attachable: null
      ca_cert: null
      client_cert: null
      client_key: null
      connected: []
      debug: false
      docker_host: unix://var/run/docker.sock
      driver: bridge
      driver_options: {}
      enable_ipv6: null
      force: false
      internal: null
      ipam_config: null
      ipam_driver: null
      ipam_driver_options: null
      labels: {}
      name: matrix
      scope: null
      ssl_version: null
      state: present
      timeout: 60
      tls: false
      tls_hostname: null
      use_ssh_client: false
      validate_certs: false
  msg: 'Error connecting: Error while fetching server API version: (''Connection aborted.'', FileNotFoundError(2, ''No such file or directo                                                                     ry''))'
PLAY RECAP ********************************************************************************************************************************
chat.domainname.com     : ok=11   changed=0    unreachable=0    failed=1    skipped=10   rescued=0    ignored=0

**Comment by sobriant74** [comment]
I see a similar case previously, https://github.com/spantaleev/matrix-docker-ansible-deploy/issues/886, but I too have the python3 mention in my hosts files and both my servers are running the latest update and upgrades for ubuntu 22.

**Comment by sobriant74** [comment]
ok, it was a server fingerprint issue. I had to clear out all the known fingerprints, run the 'ansible-playbook -i inventory/hosts setup.yml --tags=setup-all,start --ask-pass' command again from inside devture/ansible 'docker run -it --rm \
-w /work \ -v `pwd`:/work \ -v $HOME/.ssh/id_rsa:/root/.ssh/id_rsa:ro \ --entrypoint=/bin/sh \docker.io/devture/ansible:2.13.6-r0', it asked if it should load the unknonw fingerprint, I said 'yes' and then it failed quickly. I ran the command again and then it loaded the playbook on the host server. I did get a new error about "TASK [custom/matrix-nginx-proxy : Attempt initial SSL certificate retrieval with standalone authenticator " failing, but thats a new issue, so I'll close this ticket.
